apps/ios/Shared/Model/ChatModel.swift

```diff
@@ -115,6 +129,34 @@ class ItemsModel: ObservableObject {
             }
         }
     }
+
+    public var contentTag: MsgContentTag? {
+        switch secondaryIMFilter {
+        case nil: nil
+        case .groupChatScopeContext: nil
+        case let .msgContentTagContext(contentTag): contentTag
+        }
+    }
+
+    public var groupScopeInfo: GroupChatScopeInfo? {
+        switch secondaryIMFilter {
+        case nil: nil
+        case let .groupChatScopeContext(scopeInfo): scopeInfo
+        case .msgContentTagContext: nil
+        }
+    }
+}
+
+class PreloadState {
```

Comment:
moved from ChatScrollHelpers, static shared replaced with instance in ItemsModel